tests/lib/external/snapd-testing-tools/tools/os.query

```diff
@@ -73,19 +73,59 @@ is_core24() {
 }
 
 is_core_gt() {
-    compare_ubuntu "${1:-}" "-gt"
+    VERSION=$1
+    if [ -z "$VERSION" ]; then
+        echo "os.query: version id is expected"
+        exit 1
+    fi
+
+    # While ubuntu classic image is used to build a core image
+    if ! grep -qFx 'ID=ubuntu-core' /etc/os-release; then
+        VERSION="${VERSION}.04"
+    fi
+    compare_ubuntu "$VERSION" "-gt"
 }
 
 is_core_ge() {
-    compare_ubuntu "${1:-}" "-ge"
+    VERSION=$1
```

Comment:
```suggestion
    VERSION="$1"
```

Reply:
I understad this is not a shellcheck issue when the variable is a function parameter

Reply:
I was very unclear on this, as my instinct told me one thing and then shellcheck said something else.

I quickly tested this with a `good-or-bad` script:
```sh
foo() {
  ARG=$1
  echo "Arg is $ARG"
}


foo "$1"
```

Running it via `bash` did the right thing despite missing quoting:

```
zyga@ciri:~$ bash good-or-bad "one two"
Arg is one two
```

Running it via `sh` also did the right thing:
```
zyga@ciri:~$ dash good-or-bad "one two"
Arg is one two
```

So I guess my instinct is really just wrong on this one. As arguments this is fine.